### PR TITLE
Change type-hinting of Redirector and URL Generator class (pull #289)

### PR DESCRIPTION
**Pull request by kevintanjung**
Sending pull #289 to master branch instead of 5.1 branch, also remove checking for Lumen instance.

### CODE REVIEW

File: src/Routing/UrlGenerator.php [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 namespace Laravel\Lumen\Routing;
 
-use Laravel\Lumen\Application;
+use Illuminate\Contracts\Foundation\Application;
```
Comment (crynobone) [inline src/Routing/UrlGenerator.php]: `Laravel\Lumen\Application` in 5.2 is no longer implementing `Illuminate\Contracts\Foundation\Application`.

Comment (kevintanjung) [inline src/Routing/UrlGenerator.php]: @taylorotwell told me to send the pull to the master branch, I did push it to 5.1 before. So... what should I do now?

Comment (crynobone) [inline src/Routing/UrlGenerator.php]: Typehint directly to `Laravel\Lumen\Application`

**Comment by GrahamCampbell** [comment]
Please just send your code, not the other stuff too.

**Comment by kevintanjung** [comment]
Yeah, but isn't it because the `master` branch is behind 4 commits from `5.1`? Should I do my code changes on the `master` branch  then?

Moreover, as @crynobone has pointed out, the `master` branch version of `Laravel\Lumen\Application` does not implement the `Illuminate\Contract\Foundation\Application` anymore. If I type-hinted, as I suggest to `5.1`, then it will surely fail.

I don't mind doing the changes but I think we should rather wait what @taylorotwell has to say first right, before I made another PR.

**Pull request closed without merging**